xbmc/Application.cpp

```diff
@@ -3477,7 +3480,11 @@ void CApplication::OnPlayBackStarted(const CFileItem &file)
   g_pythonParser.OnPlayBackStarted(file);
 #endif
 
-  m_itemCurrentFile.reset(new CFileItem(file));
+  // reset the m_itemCurrentFile, but NOT when playing a stack (as we need it to correctly track bookmarks)
```

Comment:
what happens if a stack is playing and user starts some other file?

Reply:
Works just fine. I have tested this case and m_itemCurrentFile is already set to the new file by the time we get here. CApplication::PlayFile sets it immediately when restart = false (the default).
It seems that this code is redundant and could be dropped. Unless setting m_itemCurrentFile in PlayFile is in fact incorrect? If we say that m_itemCurrentFile should be set during playback, that is from OnPlayBackStarted to OnPlayerCloseFile (because it's asynchronous as you said in your other comment), then that needs to be changed. The helper methods that I introduced will also need some change then, because they are called in places e.g. before playback starts...

Reply:
> It seems that this code is redundant and could be dropped. Unless setting m_itemCurrentFile in PlayFile is in fact incorrect?

ahh. Settng it in PlayFile is ok. Maybe drop here if redundant and not required.